ha/setup/ha_setup.py

```diff
@@ -791,6 +793,73 @@ def process(self):
         """
         pass # TBD
 
+
+class PreUpgradeCmd(Cmd):
+    """
+    Pre-Upgrade Cmd
+    """
+    name = "pre_upgrade"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process command.
+
+        NOTE: Consul is not expected to be upgraded, so no need to backup
+        consul config
+        Configuration backup is performed for every node while pacemaker setup
+        only on one node (cluster level).
+        """
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                Log.info("Performing pre disruptive upgrade routines on cluster \
+                         level")
+                delete_resources()
+            else:
+                Log.info("Performing pre disruptive upgrade routines on node \
+                         level")
+                backup_configuration()
+        except Exception as err:
+            raise SetupError("Pre-upgrade routines failed") from err
+
+
+class PostUpgradeCmd(Cmd):
+    """
+    Post-Upgrade Cmd
+    """
+    name = "post_upgrade"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+        machine_id = self.get_machine_id()
+        self.s3_instance = UpgradeCmd.get_s3_instance(machine_id)
+
+    def process(self):
+        """
+        Process command.
+        """
+        try:
+            if os.environ['PRVSNR_MINI_LEVEL'] == 'cluster':
+                machine_id = self.get_machine_id()
+                cluster_id = Conf.get(self._index, f"server_node.{machine_id}.cluster_id")
+                mgmt_info: dict = self.get_mgmt_vip(machine_id, cluster_id)
+                node_count: int = len(self.get_nodelist(fetch_from=UpgradeCmd.HA_CONFSTORE))
+
+                Log.info(f"Performing post disruptive upgrade routines on cluster \
+                        level. cluster_id: {cluster_id}, mgmt_info: {mgmt_info}, node_count: {node_count}")
+                perform_post_upgrade(self.s3_instance, do_unstandby=False)
```

Comment:
Still we are not passing mgmt_info and node_count?

Reply:
Oops. You are right.

Reply:
Fixed.